### DISCUSSION TITLE
Closing all spans and flushing before calling `exec`?

### DISCUSSION THREAD

**Discussion by zmitchell** [tokio-rs/tracing#2915] [Q&A]
I have a CLI tool that calls `exec` and thus never returns from the function that calls `exec`. I think this is causing some of my spans to never finish (and thus telemetry to not be sent). Is there a way I can tell the `tracing` subsystem "pretend we're about to return from main and flush what you've got" before I call `exec`?

The offending function: [flox activate](https://github.com/flox/flox/blob/a560b5cfdd14990a02f61dae907fd6afa886dc6b/cli/flox/src/commands/environment.rs#L394) command with Sentry integration.

**Comment by davidbarsky** [comment]
Sorry, it took me a second to understand the question: I'm guessing that `exec` means that `Drop` isn't ever called, which is the basis of the functionality you're looking for. There's nothing like what you're describing, you might need a some mechanism on the Sentry later to force a flush, not `tracing` in general.
